lib/state/stateManager.ts

```diff
@@ -372,6 +384,13 @@ export default class StateManager {
     if (!touched) {
       throw new Error('Reverting to invalid state checkpoint failed')
     }
+    // Exceptional case due to consensus issue in Geth and Parity.
+    // See [EIP-716](https://github.com/ethereum/EIPs/issues/716) for context.
+    // The RIPEMD precompile has to remain *touched* even when the call reverts,
+    // and be considered for deletion.
+    if (this._touched.has(ripemdPrecompileAddress)) {
```

Comment:
I moved the re-touching of precompiles in case of OOG here (not a strong preference, I could re-move it to evm.ts). It is also limited to the precompile at `0x03`.